core/templates/components/forms/custom-forms-directives/thumbnail-display.component.ts

```diff
@@ -0,0 +1,49 @@
+// Copyright 2020 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controller for thumbnail display.
+ */
+
+import { Component, Input, OnInit } from '@angular/core';
+import { DomSanitizer } from '@angular/platform-browser';
+import { downgradeComponent } from '@angular/upgrade/static';
+
+@Component({
+  selector: 'thumbnail-display',
+  templateUrl: './thumbnail-display.component.html',
+  styleUrls: []
+})
+export class ThumbnailDisplayComponent implements OnInit {
+  constructor(private sanitizer: DomSanitizer) {}
+    @Input() source: string;
+    @Input() height: string;
+    @Input() width: string;
+    @Input() classes: unknown;
+    @Input() background: string;
+    imageSource = null;
+
+    ngOnInit(): void {
+      if (typeof (this.source) === 'string') {
```

Comment:
Why are we using if/ else here? If it is to include input for both SVG and normal images then maybe the type definition of `source` is wrong. Wrong is of type `string` according to your code so this if statement doesn't make sense.

Reply:
I used the if/else here to check if the image in base64 string or a object type. When uploading the image the image is loaded and displayed as base64,, which was treated as unsafe by angular. 
Thus, had to check this and add a bypass.

Reply:
Keep the types as simple as possible. I think the image data coming from service would be of the Object type. Hence I asked you to move to Bypass security line from the service to this component

Reply:
I would suggest adding another input to the component for the source type of the image. The source type should not be "guessed" here.

I would also suggest, at the place where this logic is moved, make sure that the checks are stronger. i.e. if you expect a base64 input, make sure that it is a valid base64 input, not just a random string.